src/lib/amazon.ts

```diff
@@ -50,12 +50,13 @@ export async function getFile(client: S3, bucket: string, key: string): Promise<
  */
 export async function uploadFile(client: S3, options: IUploadS3Options): Promise<IUploadS3Result> {
   try {
-    const fileStream = await readFile(options.path);
+    let fileStream;
+    if (options.path) fileStream = await readFile(options.path);
```

Comment:
readFile is asynchronous so there is a chance the stream will be empty here I believe.